kernel/clambda.ml

```diff
@@ -436,23 +439,22 @@ let check_compilable ib =
 
 let is_value lc =
   match lc with
-  | Lval _ -> true
+  | Lval _ | Lint _ -> true
   | _ -> false
 
 let get_value lc =
   match lc with
   | Lval v -> v
+  | Lint i -> val_of_int i
   | _ -> raise Not_found
 
-let mkConst_b0 n = Lval (Cbytecodes.Const_b0 n)
-
 let make_args start _end =
   Array.init (start - _end + 1) (fun i -> Lrel (Anonymous, start - i))
 
 (* Translation of constructors *)
 let expand_constructor tag nparams arity =
   let ids = Array.make (nparams + arity) Anonymous in
-  if arity = 0 then mkLlam ids (mkConst_b0 tag)
+  if arity = 0 then mkLlam ids (Lint tag)
```

Comment:
This is the part of the pull request I am the most unsure of. Saying that a nullary constructor is an integer breaks the abstraction barrier. I would have kept an indirection to `Const_b0` here. If keeping an indirection is too costly, I would prefer that another constructor be used, e.g. `Lenum`, rather than overloading `Lint` with two different meanings.

Reply:
`Lint` only means constant constructor (when primitive integers are added, they use a new `Luint` constructor). So maybe a better way is simply to change the name of the constructor.

Reply:
```
let one63 = Lint 1
```
?

Reply:
```
  | Int i -> Lint (Uint63.to_int i)
```

Reply:
Those two are wrong! They will work on 64 bit architecture, but they break an abstraction, and won't work on 32 bit architectures. Let me check again the code there.